parachain/pallets/ethereum-light-client/src/lib.rs

```diff
@@ -415,29 +457,38 @@ pub mod pallet {
 			Ok(())
 		}
 
+		fn find_finalized_decendant(
```

Comment:
Actually, this should be named `find_finalized_ancestor`, since we are looking for an ancestor of the best_block.